utests/tests/ptf/test_mask.py

```diff
@@ -53,3 +55,79 @@ def test_mask__mask_has_problem_with_conditional_fields(self):
             255,
             255,
         ], "Only gpid field should be masked"
+
+    def test_mask__conditional_field__gpid_should_be_masked_correctly(self):
+        masked_simple_vxlan = Mask(simple_vxlan_packet(vxlan_flags="G"))  # type: Mask
+        masked_simple_vxlan.set_do_not_care_packet(VXLAN, "gpid")
+        # UDP chksum will be different when we change VXLAN
+        masked_simple_vxlan.set_do_not_care_packet(UDP, "chksum")
+
+        packet_wth_custom_gpid = simple_vxlan_packet(vxlan_flags="G")  # type: Packet
+        packet_wth_custom_gpid[VXLAN].gpid = 0x15  # 21
+
+        assert masked_simple_vxlan.pkt_match(
+            packet_wth_custom_gpid
+        ), "Packets should match"
+
+    @pytest.mark.parametrize(
+        "elements_to_ignore", ((UDP, "sport"), (IP, "not_existing_field"))
+    )
+    def test_mask__negative__try_to_mask_not_existing_layer_or_field(
+        self, elements_to_ignore
+    ):
+        simple_packet = simple_ip_packet()  # type: Packet
+        masked_packet = Mask(simple_packet)  # type: Mask
+        with pytest.raises(MaskException):
+            masked_packet.set_do_not_care_packet(
+                elements_to_ignore[0], elements_to_ignore[1]
+            )
+        assert not masked_packet.valid
+
+    def test_mask__validate_str_conversion(self):
+        masked_packet = Mask(simple_vxlan_packet())  # type: Mask
+        assert str(masked_packet) == EXPECTED_MASKED_PACKET
```

Comment:
I would like an example with some fields masked.

Reply:
I changed this example to the one with a masked one field, for example, IP's checksum.